### ISSUE TITLE
any option for incremental selection

### ISSUE THREAD

**Issue by mosheavni** [atusy/treemonkey.nvim#2]
say I want to select a block using `vm`, then, I extend the visual selection to another block, and hit `m` again, I get the letters for the next block, when I choose the second block, it replaces the selection instead of adding to it.
just like this:
https://github.com/nvim-treesitter/nvim-treesitter?tab=readme-ov-file#incremental-selection
gif for example:
![2023-12-28_17-22-56 (1)](https://github.com/atusy/treemonkey.nvim/assets/17252601/8a4672a7-1bc7-4468-974f-71cf54fc0e1f)

**Comment by atusy** [comment]
Interesting idea!

I implemented a new option `action` and now you can achieve the goal by

```lua
vim.keymap.set({ "x", "o" }, "m", function()
	require("treemonkey").select({
		ignore_injections = false,
		action = require("treemonkey.actions").unite_selection,
	})
end)
```

Enjoy!

**Comment by mosheavni** [comment]
thank you kindly for the feature!
It's working only for upper selections and not lower, meaning if I want to extend up, I can, down I can't:
![2023-12-31_09-57-02 (1)](https://github.com/atusy/treemonkey.nvim/assets/17252601/e78e769d-f149-44d3-9c28-342720c29439)

**Comment by atusy** [comment]
Oops... Maybe the feature has degraded somewhere...
I remember it worked before...
Will take a look in days.

**Comment by atusy** [comment]
Should be fixed by 0a9c3ba

**Comment by mosheavni** [comment]
> Should be fixed by [0a9c3ba](https://github.com/atusy/treemonkey.nvim/commit/0a9c3ba122b4741a8329e2c1371dadf9f0e888b2)

you're the best! works great!

**Comment by mosheavni** [comment]
I added the plugin to nvim awesome list:
https://github.com/rockerBOO/awesome-neovim/pull/1023

if you think a better description fit, please let me know :)
